File: 0014-longest-common-prefix/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        shortest_len = len(min(strs, key=len))

        prefix = ""
        i = 0
        while i < shortest_len:
            for string in strs:                
                if string[i] != strs[0][i]:
                    return prefix

            prefix += string[i]
            i += 1

        return prefix
# @lc code=end

File: 0014-longest-common-prefix/14.longest-common-prefix.cs
/*
 * @lc app=leetcode id=14 lang=csharp
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        int shortestLen = strs.OrderBy(x => x.Length).FirstOrDefault().Length;

        string prefix = "";

        if (strs.Length == 1)
        {
            return strs[0];
        }
        else
        {
            for (int i = 0; i < shortestLen; i++)
            {
                foreach (string s in strs)
                {
                    if (s[i] != strs[0][i])
                    {
                        return prefix;
                    }   
                }
                prefix += strs[0][i];
            }
        }
        
        return prefix;
    }
}
// @lc code=end

